src/vector/geometry.rs

```diff
@@ -424,10 +425,30 @@ impl Geometry {
         Ok(unsafe { Geometry::with_c_geometry(new_c_geom, true) })
     }
 
+    /// Compute geometry area in square units of the spatial reference system in use.
+    ///
+    /// Supported for on`LinearRing`, `Polygon` or `MultiPolygon`.
+    /// Returns zero for all other geometry types.
+    ///
+    /// See: [`OGR_G_Area`](https://gdal.org/api/vector_c_api.html#_CPPv410OGR_G_Area12OGRGeometryH)
     pub fn area(&self) -> f64 {
         unsafe { gdal_sys::OGR_G_Area(self.c_geometry()) }
     }
 
+    /// Computes and returns the axis-aligned 2D bounding envelope for this geometry.
+    pub fn envelope(&self) -> Envelope {
+        let mut envelope = Envelope::default();
+        unsafe { gdal_sys::OGR_G_GetEnvelope(self.c_geometry(), &mut envelope) };
```

Comment:
Nit: do you want to give https://doc.rust-lang.org/stable/std/mem/union.MaybeUninit.html a try here?

Reply:
Do you mean using `MaybeUninit` instead of `Default`, and skip the initialization?

Reply:
Yes. It's not very important, just slightly nice to skip the initialization.